pyomo/core/base/constraint.py

```diff
@@ -399,8 +405,10 @@ def lb(self):
     @property
     def ub(self):
         """Access the value of the upper bound of a constraint expression."""
-        bound = value(self._ub())
-        if bound is not None and not math.isfinite(bound):
+        bound = self._ub()
+        if bound.__class__ not in native_types:
+            bound = value(bound)
+        if bound in _nonfinite_values or bound != bound:
```

Comment:
Same question as above.

Reply:
If bound is `float('nan')`, then `(bound != bound) is True`